sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1051,33 +1261,341 @@ public void populateDisplayData(DisplayData.Builder builder) {
     }
   }
 
-  ////////////////////////////////////////////////////////////////////////////////////////////////
-
-  private static final Logger LOG = LoggerFactory.getLogger(KafkaIO.class);
-
   /**
-   * Returns a new config map which is merge of current config and updates. Verifies the updates do
-   * not includes ignored properties.
+   * A {@link PTransform} to read from Kafka. See {@link KafkaIO} for more information on usage and
+   * configuration.
    */
-  private static Map<String, Object> updateKafkaProperties(
-      Map<String, Object> currentConfig,
-      Map<String, String> ignoredProperties,
-      Map<String, Object> updates) {
+  @Experimental(Kind.PORTABILITY)
+  @AutoValue
+  public abstract static class ReadAll<K, V>
```

Comment:
I really would like to avoid calling this ReadAll because it is not consistent with the ongoing work where we call ReadAll transforms the ones from `PCollection<Read>`

Reply:
Thanks for starting the discussion. If taking x-lang usage into consideration, `Read` is not a good choice as input for these DoFn.